### DISCUSSION TITLE
[qualif.] what about contributing personally support for Win32 meaning Windows 11 ?

### DISCUSSION THREAD

**Discussion by arkenidar** [koreader/koreader#12220] [Q&A]
I have noticed this project and I am using it in Android especially . on my Laptp PC : I have Windows 11 , Debian 12 , Ubuntu 24.04 , and I am a programmer that has some experience in cross-platform C/C++ , cmake, libSDL , LuaJIT with its FFI (used also for libSDL) , Love2D , DHTML5 . I possess : Google Android device , Amazon Kindle PaperWhite , both having a Linux kernel . plus laptop with latest Microsoft Windows , and Ubuntu & Debian .

So what is the current state of "win32" ? ( as I noticed some mention of it in some script ) ?

Can setup a build of the SDL emulator also for MS Windows ? I have some experience of this : SDL in Linuxes and in MS Windows from the same codebase , setup to build as widely as possible as much I can contribute .

Is the SDL emulator the part of this team effort and project the part should I want to work it to achieve a use in MS Windows environment ? BTW I used MSYS + MinGW , I would suggest and go this way for SDL on Windows . I used also CMake in MS VSCode with love2d.org introductorily , then luajit ffi projects , and C / C++ builds also related to SDL and lua / luajit / mini-lua .

Please be of kind feedback . I don't want to waste this possibility I see .

I see some mention about WSL but doing it natively using GNU tools seems to be the way !

Even if I have to port the Input / Output systems but this is within my expertise both with cross-platform sdl and luajit ffi .
I used to be a distro hopper , and I enjoy the variety and the possibilities of being able to rely on several platforms to support my needs and maybe community needs .

**Comment by Frenzie** [comment]
> So what is the current state of "win32" ? ( as I noticed some mention of it in some script ) ?

Unmaintained. SLD takes care of input/output.

See https://github.com/koreader/koreader/issues/7653 or historically https://github.com/koreader/koreader-base/pull/225 for some more information

tl;dr just compile it and fix things in the build system as you go along (and I would aggressively disable things you don't need to get a minimal working sample sooner only to reenable them later)

Probably mainly see https://github.com/koreader/koreader-base/blob/02a3ade4eb953a03dd8a277b958e48dac0ce05a4/cmake/CMakeLists.txt (and individual CMakeLists.txt in base/thirdparty)

Also pinging @benoit-pierre in case there's something else that's important to mention.
